### PR TITLE
feat: Add more sorter hooks

### PR DESCRIPTION
**Pull request by tjdevries**

### CODE REVIEW

**Comment by tami5** [comment]
:thinking: Whats an example use case of this?

**Comment by tjdevries** [comment]
@tami5 https://youtu.be/VuNIsY6JdUw <-- I think this video explains it very clearly.

**Comment by Conni2461** [comment]
@tami5 we said i need to be done till sunday evening not that i have to ship it. FYI it currently segfaults when i call it in neovim. Totally random btw. Like it calculates everything correctly but when i use neovim again, neovim will segfault after 10s. valgrind didn't help and i can't reproduce it with a c executable.

**Comment by Conni2461** [comment]
Don't listen to me :rofl: I am going crazy for an hour and as soon as i write here i find it

File: lua/telescope/sorters.lua [hunk 1]
```diff
@@ -38,13 +38,24 @@ Sorter.__index = Sorter
...
   return setmetatable({
     score = opts.score,
     state = {},
     tags = opts.tags,
+
+    -- State management
+    init    = opts.init,
+    start   = opts.start,
+    finish  = opts.finish,
+    destory = opts.destroy,
```
Comment (Conni2461) [inline lua/telescope/sorters.lua:57]: TJ you are driving me crazy right now :sob: typo here: should be `destroy`

**Comment by tami5** [comment]
> @tami5 we said i need to be done till sunday evening not that i have to ship it. FYI it currently segfaults when i call it in neovim. Totally random btw. Like it calculates everything correctly but when i use neovim again, neovim will segfault after 10s. valgrind didn't help and i can't reproduce it with a c executable.

😵🔫

File: lua/telescope/sorters.lua (hunk 1 above)
Comment (tjdevries) [reply lua/telescope/sorters.lua:57]: hahahahahahahaha my bad

**Comment by smithbm2316** [comment]
@tjdevries "fix breaking conni brain" might be the funniest commit message I've ever seen. Y'all have too much fun here sometimes and I'm here for it. That is all 🙂

**Comment by Conni2461** [comment]
I didn't laugh :| I was yelling at my pc why i leaked memory. And was searching everywhere in my code. 
Like 1500 lines of c vs 30 lines of lua code written by the telescopic johnson. Where do you expect that memory leak. And valgrind also didn't report anything. 
So yeah i almost lost my mind in 30 mins

**Comment by tami5** [comment]
> I didn't laugh :| I was yelling at my pc why i leaked memory. And was searching everywhere in my code. 
> 
> Like 1500 lines of c vs 30 lines of lua code written by the telescopic johnson. Where do you expect that memory leak. And valgrind also didn't report anything. 
> 
> So yeah i almost lost my mind in 30 mins

Tick Tock Tick Tock 😂. If it was me I'd relax and work on another interesting project then come back with a renewed mind. No need for rushing just let it go 🥲, and let me have a peaceful week next week 🤣.

**Comment by tjdevries** [comment]
@Conni2461 I can add more hooks in if you need them (since I can't allow you to type any code that lands in this repository anymore). Let me know :)

**Pull request merged**